src/cpp/common/tls_credentials_options_util.cc

```diff
@@ -0,0 +1,116 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "src/cpp/common/tls_credentials_options_util.h"
+#include <grpcpp/security/tls_credentials_options.h>
+
+namespace grpc_impl {
+namespace experimental {
+
+/** Converts the Cpp key materials to C key materials; this allocates memory for
+ * the C key materials. Note that the user must free
+ * the underlying pointer to private key and cert chain duplicates; they are not
+ * freed when the UniquePtr<char> member variables of PemKeyCertPair are unused.
+ * Similarly, the user must free the underlying pointer to c_pem_root_certs. **/
+grpc_tls_key_materials_config* ConvertToCKeyMaterialsConfig(
```

Comment:
why not expose a function in TlsKeyMaterialsConfig to extract c style KeyMaterialsConfig? e.g.,
grpc_tls_key_materials_config* TlsKeyMaterialsConfig::getKeyMaterialsConfig();

Reply:
Per our discussion, I left this function as is. (For reference, it is an internal function that is hidden from the user to try to avoid memory management issues.)